modules/dnn/CMakeLists.txt

```diff
@@ -175,12 +182,32 @@ endif()
 
 set(dnn_runtime_libs "")
 
+file(GLOB_RECURSE dnn_srcs
+     "${CMAKE_CURRENT_LIST_DIR}/src/*.cpp"
+)
+file(GLOB_RECURSE dnn_int_hdrs
+     "${CMAKE_CURRENT_LIST_DIR}/src/*.hpp"
+     "${CMAKE_CURRENT_LIST_DIR}/src/*.h"
+)
+set(dnn_plugin_srcs ${dnn_srcs} ${dnn_int_hdrs})
+ocv_list_filterout_ex(dnn_plugin_srcs
+    "/src/dnn.cpp$|/src/dnn_utils.cpp$|/src/dnn_utils.cpp$|/src/dnn_read.cpp$|/src/registry.cpp$|/src/backend.cpp$"
+    # importers
+    "/src/(caffe|darknet|onnx|tensorflow|torch)/"
+    # executors
+    "/src/(cuda|cuda4dnn|ocl4dnn|vkcom|webnn)/"
+)
+
 ocv_option(OPENCV_DNN_OPENVINO "Build with OpenVINO support (2021.4+)" (TARGET ocv.3rdparty.openvino))
 if(TARGET ocv.3rdparty.openvino AND OPENCV_DNN_OPENVINO)
   if(NOT HAVE_OPENVINO AND NOT HAVE_NGRAPH)
     message(FATAL_ERROR "DNN: Inference Engine is not supported without enabled 'nGraph'. Check build configuration.")
   endif()
-  list(APPEND dnn_runtime_libs ocv.3rdparty.openvino)
+  if("openvino" IN_LIST DNN_PLUGIN_LIST OR DNN_PLUGIN_LIST STREQUAL "all")
+    add_subdirectory(misc/plugin/openvino)  # plugin doesn't support PCH, separate directory scope is necessary
```

Comment:
Fixed.

To properly use OV plugin need to disable GAPI module or OV-in-GAPI:
- `-DBUILD_opencv_gapi=OFF`
- or `-DOPENCV_GAPI_WITH_OPENVINO=OFF`

Otherwise gapi links OpenVINO to opencv_world directly, which affects DNN code compilation flags too.